compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1107,6 +1101,7 @@ OMR::Z::TreeEvaluator::floatRemHelper(TR::Node * node, TR::CodeGenerator * cg)
       generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BE, node, labelOK);        // it is not < +0
       generateRRInstruction(cg, TR::InstOpCode::LCEBR, node, targetRegister, targetRegister); // negate answer to be -0
       generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, labelOK);        // it is not < +0
+      TR::Node::recreate(node, TR::fcall);
```

Comment:
I do not believe this is the best way to achieve what you want. When you recreate a node the underlying infrastructure will remove all non-applicable facts you know about this IL. For example if you knew this particular node was always non-negative and the node was marked as such the `recreate` API may wipe out this property since it is not applicable to calls (this is hypothetical to illustrate the point; not sure if it actually happens for the non-negative flag but surely it does for others). Then when you `recreate` again later you've lost a piece of information.

To achieve what you want here it's best to create a new node if the infrastructure for `buildSystemLinkageDispatch` does not support a variadic call where you can pass arguments directly. This way you also avoid the hairiness of the `unsetRegister` hack below.